packages/chevrotain/src/parse/cst/cst.ts

```diff
@@ -24,11 +24,45 @@ import { GAstVisitor } from "../grammar/gast/gast_visitor_public"
 import {
     CstChildrenDictionary,
     CstNode,
+    CstNodeLocation,
     IOptionallyNamedProduction,
     IProduction,
     IToken
 } from "../../../api"
 
+export function setNodeLocationOnlyOffset(
+    nodeLocation: CstNodeLocation,
+    locationInformation: any
+): void {
+    if (nodeLocation.startOffset > locationInformation.startOffset) {
+        nodeLocation.startOffset = locationInformation.startOffset
+    }
+}
+
+export function setNodeLocationOnlyStart(
+    nodeLocation: CstNodeLocation,
+    locationInformation: any
+): void {
+    if (nodeLocation.startOffset > locationInformation.startOffset) {
```

Comment:
we are parsing left to right. so if we ignore error recovery scenarios
this condition can only be entered **once** (for the first token/subrule) of the CSTNode.
Maybe we should leave this for a future optimization though because
this adds complexity with handling error recovery Tokens.